xsub.go

```diff
@@ -72,7 +88,64 @@ func (xsub *xsubSocket) GetOption(name string) (interface{}, error) {
 
 // SetOption is used to set an option for a socket.
 func (xsub *xsubSocket) SetOption(name string, value interface{}) error {
-	return xsub.sck.SetOption(name, value)
+	err := xsub.sck.SetOption(name, value)
+	if err != nil {
+		return err
+	}
+
+	var (
+		topic []byte
+	)
+
+	switch name {
+	case OptionSubscribe:
+		k := value.(string)
+		xsub.subscribe(k, 1)
+		topic = append([]byte{1}, k...)
+
+	case OptionUnsubscribe:
+		k := value.(string)
+		topic = append([]byte{0}, k...)
+		xsub.subscribe(k, 0)
+
+	default:
+		return ErrBadProperty
+	}
+
+	xsub.sck.mu.RLock()
```

Comment:
could you add a comment on why we take the lock here?

Reply:
I copied this bit from sub.go in master - see line 110. I do not know why it was there in the first place; I was just copying the existing subscribe logic, and didn't want to change anything that didn't seem necessary.

Reply:
ok.
seems like I'd need somebody else to review _my_ PRs :)